discovery/pkg/sync/queue.go

```diff
@@ -110,17 +109,21 @@ func (sq *Queue) processNextWorkItem() bool {
 	action, ok := obj.(Action)
 	if !ok {
 		sq.Workqueue.Forget(obj)
-		sq.Metrics.QueueSizeGaugeMetric(sq.BackendName, sq.ClusterType, sq.Workqueue.Len())
+		sq.Metrics.QueueSizeGaugeMetric(sq.Workqueue.Len())
 		sq.Logger.Errorf("got an unknown item of type %T in the queue", obj)
 		return true
 	}
 
-	err := action.Sync(sq.KubeClient, sq.Metrics, sq.BackendName)
+	err := action.Sync(sq.KubeClient, sq.Logger)
+	if err != nil {
+		action.SetMetricError(sq.Metrics)
+	}
+	action.SetMetrics(sq.KubeClient, sq.Metrics, sq.Logger)
```

Comment:
I am wondering how these metrics will behave if we need to retry. Is it OK to set these metrics multiple times for the same service because of retries? Will open a new issue to discuss/look into this.